hypothesis_bio/hypothesis_bio.py

```diff
@@ -176,44 +177,112 @@ def fasta(draw) -> str:
 
 
 @composite
-def fastq_quality(draw: Callable, size=0) -> str:
+def sequence_id(
+    draw, blacklist_characters: Sequence[str] = ">@", max_size: int = 100
+) -> str:
+    """Generates a sequence ID.
+
+    Arguments:
+    - `blacklist_character`: Characters to not include in the sequence ID.
+    - `max_size`: Maximum length of the sequence ID.
+    """
+    return draw(
+        text(
+            alphabet=characters(
+                blacklist_characters=blacklist_characters,
+                min_codepoint=33,
+                max_codepoint=MAX_ASCII,
+            ),
+            max_size=max_size,
+        )
+    )
+
+
+@composite
+def fastq_quality(
+    draw, size=0, min_score: int = 0, max_score: int = 62, offset: int = 64
+) -> str:
     """Generates the quality string for the FASTQ format
 
     Arguments:
     - `size`: Size of the quality string to be generated
+    - `min_score`: Lowest quality (PHRED) score to use.
+    - `max_score`: Highest quality (PHRED) score to use.
+    - `offset`: ASCII encoding offset.
+    for more details.
 
     Note:
-    According to https://en.wikipedia.org/wiki/FASTQ_format,
-    the range of characters for the quality string ranges from
-    byte value 0x21 to 0x7E.
+        See https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217/ for more details on
+        the quality score encoding.
     """
+    min_codepoint = min_score + offset
+    max_codepoint = max_score + offset
+
+    if max_codepoint > MAX_ASCII:
+        raise ValueError(
+            "{} is larger than the maximum ASCII value {}".format(
+                max_codepoint, MAX_ASCII
+            )
+        )
+
     return draw(
         text(
-            alphabet=characters(min_codepoint=33, max_codepoint=126),
+            alphabet=characters(
+                min_codepoint=min_codepoint, max_codepoint=max_codepoint
+            ),
             min_size=size,
             max_size=size,
         )
     )
 
 
 @composite
-def fastq(draw: Callable, fasta_source=Dict[str, str]) -> str:
+def fastq(
+    draw,
+    size=0,
+    min_score: int = 0,
+    max_score: int = 62,
+    offset: int = 64,
+    add_comment: bool = False,
+    additional_description: bool = False,
+) -> str:
     """Generate strings representing sequences in FASTQ format.
+
+    Arguments:
+    - `size`: Size of the sequence and quality string.
+    - `min_score`: Lowest quality (PHRED) score to use.
+    - `max_score`: Highest quality (PHRED) score to use.
+    - `offset`: ASCII encoding offset for quality string.
+    - `add_comment`: Add a comment string after the sequence ID, separated by a space.
+    - `additional_description`: Add sequence ID and comment after `+` on thrid line.
+    for more details.
```

Comment:
ibid

Reply:
?

Reply:
Sorry if that wasn't clear. What is the `for more details.` line doing there?